modules/homepage/src/main/frontend/src/themePage/PageStart/PageStart.jsx

```diff
@@ -0,0 +1,118 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useEffect, useState } from "react";
+import { loadExtensions } from "../../uiextension/extensionManager";
+import { Button } from "@material-ui/core";
+
+export default function PageStart(props) {
+
+  const [ componentHeights, setComponentHeights ] = useState([]);
+  const [ componentPositions, setComponentPositions ] = useState([]);
+  const [ triggerRedraw, setTriggerRedraw ] = useState(false);
+  const [ extensionData, setExtensionData ] = useState(null);
+  const [ isInitialized, setIsInitialized ] = useState(false);
+
+  const arrayEquals = (a, b) => {
+    if (a.length != b.length) {
+      return false;
+    }
+    for (let i = 0; i < a.length; i++) {
+      if (a[i] != b[i]) {
+        return false;
+      }
+    }
+    return true;
+  };
```

Comment:
This is shorter and more "modern":

```suggestion
  const arrayEquals(a, b) {
    return
      Array.isArray(a) && Array.isArray(b) &&
      a.length === b.length &&
      a.every((val, i) => val === b[i]);
  };
```